sdk/src/contexts/cosmos/context.ts

```diff
@@ -434,19 +529,45 @@ export class CosmosContext<
     };
   }
 
-  private async getCosmWasmClient(
+  private async getQueryClient(
     chain: ChainId | ChainName,
-  ): Promise<CosmWasmClient> {
+  ): Promise<QueryClient & IbcExtension> {
+    const tmClient = await this.getTmClient(chain);
+    return QueryClient.withExtensions(tmClient, setupIbcExtension);
+  }
+
+  private async getTmClient(
+    chain: ChainId | ChainName,
+  ): Promise<Tendermint34Client | Tendermint37Client> {
     const name = this.context.toChainName(chain);
     if (CosmosContext.CLIENT_MAP[name]) {
       return CosmosContext.CLIENT_MAP[name];
     }
 
-    const rpc = this.context.conf.rpcs[name];
+    const rpc = this.context.conf.rpcs[this.context.toChainName(chain)];
```

Comment:
Can just use name variable defined above

Reply:
Missed that. Thanks!